Implement event filtering for self-profiler.

michaelwoerister opened the pull request:
This is a first sketch for event filtering in the self-profiler, something that we'll want in order to keep profiling overhead low in the common case. The PR contains the commits from https://github.com/rust-lang/rust/pull/59515 and, for the moment, is meant for performance testing.

r? @wesleywiser

michaelwoerister commented:
@bors try

bors commented:
:hourglass: Trying commit 92f6145c65574b9544480fb735468811ef305140 with merge 9dd2a9910ec8832159fe9b23a1d08545aa453df3...
<!-- homu: {"type":"TryBuildStarted","head_sha":"92f6145c65574b9544480fb735468811ef305140","merge_sha":"9dd2a9910ec8832159fe9b23a1d08545aa453df3"} -->

wesleywiser commented on src/librustc/util/profiling.rs:
```diff
@@ -27,18 +27,17 @@ pub enum ProfileCategory {
+        const INCR_CACHE_LOADS   = 0b00010000;
+
+        const DEFAULT = Self::GENERIC_ACTIVITIES.bits |
+                        Self::QUERY_PROVIDERS.bits;
```
When parallel rustc becomes the default, adding `Self::QUERY_BLOCKED` might also make sense.

michaelwoerister replied:
Yes, I agree.

bors commented:
:broken_heart: Test failed - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/108118356)
<!-- homu: {"type":"TryBuildFailed","builder_url":"https://travis-ci.com/rust-lang/rust/builds/108118356","builder_name":"checks-travis"} -->

rust-highfive commented:
The job `dist-x86_64-linux` of your PR [failed on Travis](https://travis-ci.com/rust-lang/rust/jobs/192490610) ([raw log](https://api.travis-ci.com/v3/job/192490610/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
travis_fold:end:services

travis_fold:start:git.checkout
travis_time:start:0102f92c
$ git clone --depth=2 --branch=try https://github.com/rust-lang/rust.git rust-lang/rust
---
[00:06:05]   Downloaded macro-utils v0.1.2
[00:06:05]   Downloaded rustc-ap-rustc_cratesio_shim v407.0.0
[00:06:05]   Downloaded dlmalloc v0.1.3
[00:06:05]   Downloaded byte-tools v0.2.0
[00:06:05]   Downloaded measureme v0.2.0
[00:06:06]   Downloaded url_serde v0.2.0
[00:06:06]   Downloaded rustc-ap-rustc_errors v407.0.0
[00:06:06]   Downloaded idna v0.1.5
[00:06:06]   Downloaded rustc-ap-graphviz v407.0.0
---
[00:46:36]    Compiling rustc_asan v0.0.0 (/checkout/src/librustc_asan)
[00:46:37]    Compiling rustc_msan v0.0.0 (/checkout/src/librustc_msan)
[00:47:20] thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 30, kind: Other, message: "Read-only file system" }', src/libcore/result.rs:997:5
[00:47:20] note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
[00:47:20] thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 30, kind: Other, message: "Read-only file system" }', src/libcore/result.rs:997:5
[00:47:20] stack backtrace:
[00:47:20]    0:     0x7fbf496c05d3 - std::sys::unix::backtrace::tracing::imp::unwind_backtrace::h5ee153a5708476ff
[00:47:20]                                at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
[00:47:20]    1:     0x7fbf496b845b - std::sys_common::backtrace::_print::h37f7cb19e8925dd7
[00:47:20]                                at src/libstd/sys_common/backtrace.rs:71
[00:47:20]    2:     0x7fbf496bc836 - std::panicking::default_hook::{{closure}}::h71d76408e7ea4aca
[00:47:20]                                at src/libstd/sys_common/backtrace.rs:59
[00:47:20]                                at src/libstd/panicking.rs:197
[00:47:20]    3:     0x7fbf496bc5c9 - std::panicking::default_hook::hba712c5213afcfac
[00:47:20]                                at src/libstd/panicking.rs:211
[00:47:20]    4:     0x7fbf46cfa6a0 - rustc::util::common::panic_hook::h51146904b6d9df8c
[00:47:20]    5:     0x7fbf496bd028 - std::panicking::rust_panic_with_hook::h7c9d916ecd443dd9
[00:47:20]                                at src/libstd/panicking.rs:478
[00:47:20]    6:     0x7fbf496bcac1 - std::panicking::continue_panic_fmt::hfc217f9901983873
[00:47:20]                                at src/libstd/panicking.rs:381
[00:47:20]    7:     0x7fbf496bc9a5 - rust_begin_unwind
[00:47:20]                                at src/libstd/panicking.rs:308
[00:47:20]    8:     0x7fbf496e585c - core::panicking::panic_fmt::h22fbbae22ebba188
[00:47:20]                                at src/libcore/panicking.rs:85
[00:47:20]    9:     0x7fbf46e22d3d - core::result::unwrap_failed::h79406567cea1657e
[00:47:20]   10:     0x7fbf46e256ec - <measureme::mmap_serialization_sink::MmapSerializationSink as core::ops::drop::Drop>::drop::h40893ebac3478089
[00:47:20]   11:     0x7fbf499e0937 - <alloc::sync::Arc<T>>::drop_slow::hb4d08bb4dfa64c02
[00:47:20]   12:     0x7fbf499e45db - core::ptr::real_drop_in_place::h73d4610e58258978
[00:47:20]   13:     0x7fbf499e0c01 - <alloc::sync::Arc<T>>::drop_slow::hd297c215ac02409b
[00:47:20]   14:     0x7fbf499fbb7b - <alloc::rc::Rc<T> as core::ops::drop::Drop>::drop::h95229cc39ffaece8
[00:47:20]   15:     0x7fbf499b287b - core::ptr::real_drop_in_place::ha484dce8506c85f2
[00:47:20]   16:     0x7fbf499ac91e - rustc_interface::interface::run_compiler_in_existing_thread_pool::had57937cf0490b4d
[00:47:20]   17:     0x7fbf49978bd5 - <std::thread::local::LocalKey<T>>::with::h99363ecc34303cb6
[00:47:20]   18:     0x7fbf499e1894 - <scoped_tls::ScopedKey<T>>::set::h9bc93ac0f84ec0ec
[00:47:20]   19:     0x7fbf49a13b91 - syntax::with_globals::hb9fcbdbd3e7f1ce2
[00:47:20]   20:     0x7fbf4997a6d4 - std::sys_common::backtrace::__rust_begin_short_backtrace::hbf97b5db70026ac3
[00:47:20]   21:     0x7fbf496cdd29 - __rust_maybe_catch_panic
[00:47:20]                                at src/libpanic_unwind/lib.rs:87
[00:47:20]   22:     0x7fbf4997c878 - core::ops::function::FnOnce::call_once::{{vtable.shim}}::hfb1cfde5deacd290
[00:47:20]   23:     0x7fbf4969f77e - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h3b71c972b364448f
[00:47:20]                                at /rustc/9dd2a9910ec8832159fe9b23a1d08545aa453df3/src/liballoc/boxed.rs:702
[00:47:20]   24:     0x7fbf496cca8f - std::sys::unix::thread::Thread::new::thread_start::hc1cf3874635d723f
[00:47:20]                                at /rustc/9dd2a9910ec8832159fe9b23a1d08545aa453df3/src/liballoc/boxed.rs:702
[00:47:20]                                at src/libstd/sys_common/thread.rs:14
[00:47:20]                                at src/libstd/sys/unix/thread.rs:80
[00:47:20]   25:     0x7fbf4943e83c - start_thread
[00:47:20]   26:     0x7fbf48da6fdc - clone
[00:47:20] thread panicked while panicking. aborting.
[00:47:21] warning:   flock(fd, LOCK_EX);
[00:47:21] warning:   ^
[00:47:21] warning: /cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.10/compiler-rt/lib/profile/InstrProfilingUtil.c:153:3: warning: implicit declaration of function 'flock' is invalid in C99 [-Wimplicit-function-declaration]
[00:47:21] warning:   flock(fd, LOCK_UN);
[00:47:21] warning:   flock(fd, LOCK_UN);
[00:47:21] warning:   ^
[00:47:21] warning: 2 warnings generated.
[00:48:12] error: Could not compile `core`.
[00:48:12] 
[00:48:12] Caused by:
[00:48:12]   process didn't exit successfully: `/checkout/obj/build/bootstrap/debug/rustc --crate-name core src/libcore/lib.rs --color always --error-format json --crate-type lib --emit=dep-info,link -C opt-level=2 -C metadata=a767b18e8ef14871 -C extra-filename=-a767b18e8ef14871 --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -C linker=clang -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/release/deps` (signal: 4, SIGILL: illegal instruction)
[00:48:13] error: build failed
[00:48:13] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "--target" "x86_64-unknown-linux-gnu" "-j" "4" "--release" "--locked" "--color" "always" "--features" "panic-unwind backtrace profiler" "--manifest-path" "/checkout/src/libstd/Cargo.toml" "--message-format" "json"
[00:48:13] expected success, got: exit code: 101
[00:48:13] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap dist --host x86_64-unknown-linux-gnu --target x86_64-unknown-linux-gnu
---
travis_time:end:07d6d405:start=1555076585473368337,finish=1555076585484230647,duration=10862310
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:2dc810f0
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:start:crashlog
obj/cores/core.4698.!checkout!obj!build!x86_64-unknown-linux-gnu!stage1!bin!rustc
[New LWP 4703]
[New LWP 4698]
warning: Could not load shared library symbols for 7 libraries, e.g. /lib64/libpthread.so.0.
Use the "info sharedlibrary" command to see the complete listing.
Do you need "set solib-search-path" or "set sysroot"?
Core was generated by `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/rustc --crate-name core'.
Program terminated with signal SIGILL, Illegal instruction.
#0  0x00007fbf496bce4a in rust_panic_with_hook () at /rustc/9dd2a9910ec8832159fe9b23a1d08545aa453df3/src/libcore/fmt/mod.rs:316
316 /rustc/9dd2a9910ec8832159fe9b23a1d08545aa453df3/src/libcore/fmt/mod.rs: No such file or directory.
[Current thread is 1 (LWP 4703)]
#0  0x00007fbf496bce4a in rust_panic_with_hook () at /rustc/9dd2a9910ec8832159fe9b23a1d08545aa453df3/src/libcore/fmt/mod.rs:316
#1  0x00007fbf496bcac2 in continue_panic_fmt () at src/libstd/panicking.rs:381
#2  0x00007fbf496bc9a6 in rust_begin_unwind () at src/libstd/panicking.rs:308
#3  0x00007fbf496e585d in panic_fmt () at src/libcore/panicking.rs:85
#4  0x00007fbf46e22d3e in core::result::unwrap_failed::h79406567cea1657e () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/../lib/librustc-a993507cc2ce9b19.so
#5  0x00007fbf46e256ed in _$LT$measureme..mmap_serialization_sink..MmapSerializationSink$u20$as$u20$core..ops..drop..Drop$GT$::drop::h40893ebac3478089 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/../lib/librustc-a993507cc2ce9b19.so
#6  0x00007fbf499e0938 in _$LT$alloc..sync..Arc$LT$T$GT$$GT$::drop_slow::hb4d08bb4dfa64c02 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#7  0x00007fbf499e45dc in core::ptr::real_drop_in_place::h73d4610e58258978 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#8  0x00007fbf499e0c02 in _$LT$alloc..sync..Arc$LT$T$GT$$GT$::drop_slow::hd297c215ac02409b () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#9  0x00007fbf499fbb7c in _$LT$alloc..rc..Rc$LT$T$GT$$u20$as$u20$core..ops..drop..Drop$GT$::drop::h95229cc39ffaece8 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#10 0x00007fbf499b287c in core::ptr::real_drop_in_place::ha484dce8506c85f2 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#11 0x00007fbf499ac91f in rustc_interface::interface::run_compiler_in_existing_thread_pool::had57937cf0490b4d () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#12 0x00007fbf49978bd6 in _$LT$std..thread..local..LocalKey$LT$T$GT$$GT$::with::h99363ecc34303cb6 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#13 0x00007fbf499e1895 in _$LT$scoped_tls..ScopedKey$LT$T$GT$$GT$::set::h9bc93ac0f84ec0ec () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#14 0x00007fbf49a13b92 in syntax::with_globals::hb9fcbdbd3e7f1ce2 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#15 0x00007fbf4997a6d5 in std::sys_common::backtrace::__rust_begin_short_backtrace::hbf97b5db70026ac3 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#16 0x00007fbf496cdd2a in __rust_maybe_catch_panic () at src/libpanic_unwind/lib.rs:87
#17 0x00007fbf4997c879 in core::ops::function::FnOnce::call_once::_$u7b$$u7b$vtable.shim$u7d$$u7d$::hfb1cfde5deacd290 () from ./checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/../lib/librustc_driver-e0faff33e0439a04.so
#18 0x00007fbf4969f77f in call_once<(),FnBox<()>> () at /rustc/9dd2a9910ec8832159fe9b23a1d08545aa453df3/src/liballoc/boxed.rs:702
#19 0x00007fbf496cca90 in call_once<(),alloc::boxed::Box<FnBox<()>>> () at /rustc/9dd2a9910ec8832159fe9b23a1d08545aa453df3/src/liballoc/boxed.rs:702
#20 start_thread () at src/libstd/sys_common/thread.rs:14
#21 thread_start () at src/libstd/sys/unix/thread.rs:80
#22 0x00007fbf4943e83d in ?? ()
#23 0x0000000000000000 in ?? ()
travis_time:end:2dc810f0:start=1555076585490218113,finish=1555076587876852922,duration=2386634809
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:040c612e
travis_time:start:040c612e
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:018cb784
$ dmesg | grep -i kill
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

wesleywiser commented on src/librustc/util/profiling.rs:
```diff
@@ -55,7 +55,7 @@ pub struct SelfProfiler {
 
 impl SelfProfiler {
-    pub fn new() -> Result<SelfProfiler, Box<dyn Error>> {
+    pub fn new(event_filters: &Option<Vec<String>>) -> Result<SelfProfiler, Box<dyn Error>> {
```
Were you thinking of making [this change](https://rust-lang.zulipchat.com/#narrow/stream/187831-t-compiler.2Fwg-self-profile/topic/event.20filter.20.2F.20level.20of.20detail/near/163101424) in this PR as well?

michaelwoerister replied:
Eventually, but I'd like to get a perf.rlo run before and after to see if it even makes a difference.

wesleywiser replied:
Agreed!

michaelwoerister commented:
@bors try

bors commented:
:hourglass: Trying commit cb7be41876e12e08457f3ecba6a39c6ccc09f745 with merge 34ebab37974c08bbce638c049dfcdf067b47f438...
<!-- homu: {"type":"TryBuildStarted","head_sha":"cb7be41876e12e08457f3ecba6a39c6ccc09f745","merge_sha":"34ebab37974c08bbce638c049dfcdf067b47f438"} -->

bors commented:
:broken_heart: Test failed - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/108145016)
<!-- homu: {"type":"TryBuildFailed","builder_url":"https://travis-ci.com/rust-lang/rust/builds/108145016","builder_name":"checks-travis"} -->

rust-highfive commented:
The job `dist-x86_64-linux` of your PR [failed on Travis](https://travis-ci.com/rust-lang/rust/jobs/192545236) ([raw log](https://api.travis-ci.com/v3/job/192545236/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
travis_fold:end:services

travis_fold:start:git.checkout
travis_time:start:04d740e0
$ git clone --depth=2 --branch=try https://github.com/rust-lang/rust.git rust-lang/rust
---
############################################################              84.6%
######################################################################## 100.0%
[00:04:49] extracting /checkout/obj/build/cache/2019-03-20/cargo-beta-x86_64-unknown-linux-gnu.tar.gz
[00:04:49]     Updating crates.io index
[00:05:06] error: the lock file /checkout/Cargo.lock needs to be updated but --locked was passed to prevent this
[00:05:06] Build completed unsuccessfully in 0:00:31
[00:05:06] make: *** [prepare] Error 1
[00:05:07] Command failed. Attempt 2/5:
[00:05:08]     Updating crates.io index
[00:05:08]     Updating crates.io index
[00:05:08] error: the lock file /checkout/Cargo.lock needs to be updated but --locked was passed to prevent this
[00:05:08] Build completed unsuccessfully in 0:00:00
[00:05:08] make: *** [prepare] Error 1
[00:05:10] Command failed. Attempt 3/5:
[00:05:10]     Updating crates.io index
[00:05:10]     Updating crates.io index
[00:05:11] error: the lock file /checkout/Cargo.lock needs to be updated but --locked was passed to prevent this
[00:05:11] Build completed unsuccessfully in 0:00:00
[00:05:11] make: *** [prepare] Error 1
[00:05:14] Command failed. Attempt 4/5:
[00:05:14]     Updating crates.io index
[00:05:14]     Updating crates.io index
[00:05:15] error: the lock file /checkout/Cargo.lock needs to be updated but --locked was passed to prevent this
[00:05:15] Build completed unsuccessfully in 0:00:00
[00:05:15] make: *** [prepare] Error 1
[00:05:19] Command failed. Attempt 5/5:
[00:05:19]     Updating crates.io index
[00:05:19]     Updating crates.io index
[00:05:20] error: the lock file /checkout/Cargo.lock needs to be updated but --locked was passed to prevent this
[00:05:20] Build completed unsuccessfully in 0:00:00
[00:05:20] make: *** [prepare] Error 1
[00:05:20] The command has failed after 5 attempts.
travis_time:end:110c9f2c:start=1555083772692554127,finish=1555084093644517622,duration=320951963495
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

wesleywiser commented:
@michaelwoerister I rebased your branch on top of #59515 which I've updated to use `measureme@0.2.1`.

@bors try

bors commented:
:hourglass: Trying commit 10b43bb73a145adea45341a21e4cea1f98fd20a7 with merge 30ce13219eef2589a769b5d694bdc1c0aec0befc...
<!-- homu: {"type":"TryBuildStarted","head_sha":"10b43bb73a145adea45341a21e4cea1f98fd20a7","merge_sha":"30ce13219eef2589a769b5d694bdc1c0aec0befc"} -->

bors commented:
:sunny: Try build successful - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/108202698)
Build commit: 30ce13219eef2589a769b5d694bdc1c0aec0befc
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-travis":"https://travis-ci.com/rust-lang/rust/builds/108202698"},"merge_sha":"30ce13219eef2589a769b5d694bdc1c0aec0befc"} -->

kennytm commented:
@rust-timer build 30ce13219eef2589a769b5d694bdc1c0aec0befc

rust-timer commented:
Success: Queued 30ce13219eef2589a769b5d694bdc1c0aec0befc with parent 99da733f7f38ce8fe68453e859b7ac96bf7caa0f, [comparison URL](https://perf.rust-lang.org/compare.html?start=99da733f7f38ce8fe68453e859b7ac96bf7caa0f&end=30ce13219eef2589a769b5d694bdc1c0aec0befc).

rust-timer commented:
Finished benchmarking try commit 30ce13219eef2589a769b5d694bdc1c0aec0befc

wesleywiser commented:
Wow, that actually seems pretty good!

michaelwoerister commented:
Yeah, that looks like a good start.

michaelwoerister commented:
@bors try

bors commented:
:hourglass: Trying commit 9900f3fbb201ce10804b84d571af520450e9f239 with merge 4dd3271ce58c3080ddcc1bf21df704efad2b73c1...
<!-- homu: {"type":"TryBuildStarted","head_sha":"9900f3fbb201ce10804b84d571af520450e9f239","merge_sha":"4dd3271ce58c3080ddcc1bf21df704efad2b73c1"} -->

bors commented:
:sunny: Try build successful - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/108345411)
Build commit: 4dd3271ce58c3080ddcc1bf21df704efad2b73c1
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-travis":"https://travis-ci.com/rust-lang/rust/builds/108345411"},"merge_sha":"4dd3271ce58c3080ddcc1bf21df704efad2b73c1"} -->

michaelwoerister commented:
@rust-timer build 4dd3271ce58c3080ddcc1bf21df704efad2b73c1

rust-timer commented:
Finished benchmarking try commit 4dd3271ce58c3080ddcc1bf21df704efad2b73c1

michaelwoerister commented:
That last commit improves performance ever so slightly but I think I'm going to revert it again, for now, because it adds some churn and might regress the case where profiling is disabled. Not quite worth the trouble at this point (also, probably the wrong place to optimize since most of work should happen if something is recorded and not if something is filtered out).

michaelwoerister commented:
I cleaned up the implementation and added blocking of queries and incr. cache loading to the default event set. Let's see if we can get away with that performance-wise.

@bors try

bors commented:
:hourglass: Trying commit 6c3fa32e7fc0a452cf53121568593d5cc09a363f with merge 5539376de500270af54c7741ff8075316d950caf...
<!-- homu: {"type":"TryBuildStarted","head_sha":"6c3fa32e7fc0a452cf53121568593d5cc09a363f","merge_sha":"5539376de500270af54c7741ff8075316d950caf"} -->

bors commented:
:sunny: Try build successful - [checks-travis](https://travis-ci.com/rust-lang/rust/builds/108672704)
Build commit: 5539376de500270af54c7741ff8075316d950caf
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-travis":"https://travis-ci.com/rust-lang/rust/builds/108672704"},"merge_sha":"5539376de500270af54c7741ff8075316d950caf"} -->

michaelwoerister commented:
@rust-timer build 5539376de500270af54c7741ff8075316d950caf

rust-timer commented:
Finished benchmarking try commit 5539376de500270af54c7741ff8075316d950caf

michaelwoerister commented:
I think this is ready for review. Overhead looks about the same with the new event types added to the default set (slightly worse). Probably still a bit too much to enable this for all perf.rlo runs but we can look into that later.

r? @wesleywiser

rust-highfive commented:
The job `x86_64-gnu-llvm-6.0` of your PR [failed on Travis](https://travis-ci.com/rust-lang/rust/jobs/194003524) ([raw log](https://api.travis-ci.com/v3/job/194003524/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
travis_time:end:1bd86231:start=1555593053041461802,finish=1555593053795840884,duration=754379082
$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout

Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
$ export SCCACHE_BUCKET=rust-lang-ci-sccache2
$ export SCCACHE_REGION=us-west-1
$ export GCP_CACHE_BUCKET=rust-lang-ci-cache
$ export AWS_ACCESS_KEY_ID=AKIA46X5W6CZEJZ6XT55
---

[00:03:33] travis_fold:start:tidy
travis_time:start:tidy
tidy check
[00:03:34] tidy error: /checkout/src/librustc/util/profiling.rs:193: line longer than 100 chars
[00:03:35] some tidy checks failed
[00:03:35] 
[00:03:35] 
[00:03:35] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/tidy" "/checkout/src" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "--no-vendor" "--quiet"
[00:03:35] 
[00:03:35] 
[00:03:35] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test src/tools/tidy
[00:03:35] Build completed unsuccessfully in 0:00:44
[00:03:35] Build completed unsuccessfully in 0:00:44
[00:03:35] make: *** [tidy] Error 1
[00:03:35] Makefile:67: recipe for target 'tidy' failed
The command "stamp sh -x -c "$RUN_SCRIPT"" exited with 2.
travis_time:start:027e3319
$ date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
Thu Apr 18 13:14:40 UTC 2019
---
travis_time:end:0059ddce:start=1555593281325230376,finish=1555593281329617017,duration=4386641
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:0b7391b4
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb --batch -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:191fda1a
travis_time:start:191fda1a
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:19fb2708
$ dmesg | grep -i kill
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

wesleywiser commented:
That didn't seem to work. I guess bors doesn't see review comments?

@bors r+

bors commented:
:pushpin: Commit 08efbac758aa75f710b8018d8974bde5c7a149c1 has been approved by `wesleywiser`

<!-- @bors r=wesleywiser 08efbac758aa75f710b8018d8974bde5c7a149c1 -->

bors commented:
:hourglass: Testing commit 08efbac758aa75f710b8018d8974bde5c7a149c1 with merge 9ae856c4d802aed5c1663a6837b69ed3c39853f5...
<!-- homu: {"type":"BuildStarted","head_sha":"08efbac758aa75f710b8018d8974bde5c7a149c1","merge_sha":"9ae856c4d802aed5c1663a6837b69ed3c39853f5"} -->

Centril commented:
@bors retry

bors commented:
:hourglass: Testing commit 08efbac758aa75f710b8018d8974bde5c7a149c1 with merge c9e8a274f30eadd5a3cae0069d922e75e67bee8d...
<!-- homu: {"type":"BuildStarted","head_sha":"08efbac758aa75f710b8018d8974bde5c7a149c1","merge_sha":"c9e8a274f30eadd5a3cae0069d922e75e67bee8d"} -->

Centril commented:
@bors retry

bors commented:
:hourglass: Testing commit 08efbac758aa75f710b8018d8974bde5c7a149c1 with merge 271c8cfeca93e207e6be4d475ff15c9409f66917...
<!-- homu: {"type":"BuildStarted","head_sha":"08efbac758aa75f710b8018d8974bde5c7a149c1","merge_sha":"271c8cfeca93e207e6be4d475ff15c9409f66917"} -->

Centril commented:
@bors retry

bors commented:
:hourglass: Testing commit 08efbac758aa75f710b8018d8974bde5c7a149c1 with merge 316a391dcb7d66dc25f1f9a4ec9d368ef7615005...
<!-- homu: {"type":"BuildStarted","head_sha":"08efbac758aa75f710b8018d8974bde5c7a149c1","merge_sha":"316a391dcb7d66dc25f1f9a4ec9d368ef7615005"} -->

marieell commented on src/librustc/util/profiling.rs:
```diff
@@ -66,8 +82,38 @@ impl SelfProfiler {
+                    EVENT_FILTERS_BY_NAME.iter()
+                                         .map(|&(name, _)| name.to_string())
+                                         .collect::<Vec<_>>()
+                                         .join(", "));
```
I think this warrants being an error instead of just a warning.
